src/mem_table.rs

```diff
@@ -36,7 +37,10 @@ pub(crate) struct ImmutableWal {
 
 type MemTableRange<'a> = Range<'a, Bytes, (Bound<Bytes>, Bound<Bytes>), Bytes, ValueWithAttributes>;
 
-pub struct MemTableIterator<'a>(MemTableRange<'a>);
+pub struct MemTableIterator<'a> {
+    range: MemTableRange<'a>,
+    last_seq: Arc<AtomicU64>,
```

Comment:
What's the reason for incrementing seq num on read instead of on write? If I read this correctly, basically every range scan over memtable will bump seq nums.

I always had a feeling that we'll need to support multiple versions of the same key (for merge operator and most likely also for transactions). This approach seems to make it harder, because we'll need to maintain ordering by seq num.

Reply:
Yeah, this was a quick implementation that doesn't do what's supposed to. I'll change with yours and @flaneur2020's suggestions soon.

Reply:
I've made this PR a draft.